### DISCUSSION TITLE
Uncaught Error: Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by sebpowell** [vercel/next.js#47807] [Help]
### Summary

Running into a similar issue to what is described [here](https://github.com/vercel/next.js/discussions/36754), and for the life of me I can't work out what the problem is. 

Here's my markup:

```TSX
export const Component = () => {
  const formik = useFormik({
    initialValues: {
      email: "",
    },
    validationSchema: validationSchema,
  });

  return (
    <form>
      <input
        name="email"
        value={formik.values.email}
        onChange={formik.handleChange}
     />
   </form>
 )
}
```

Whenever I load the page, I get the following:

<img width="912" alt="Screenshot 2023-04-02 at 01 01 15" src="https://user-images.githubusercontent.com/1786366/229323437-b2ceb2f0-cd19-408c-b6fc-ec7d3d5e3736.png">

This file is located inside a file, which is imported into the `/app` directory (Next 13.2.1), like so:

```
/app
  /page.tsx 
/components
  /component-file.tsx
```

The ticket mentioned above seems to suggest the issue may be down to invalid markup, but as far as I can tell, the markup I'm using is valid. Any help greatly appreciated.

**Comment by sebpowell** [comment]
I've been experimenting some more with this today, and think I've worked out what is going on. I use the NordPass browser extension for password management. Said extension (like others no doubt), seems to look for any input that has the word 'email' contained within the 'name' attribute. If it finds one, it attaches additional attributes, that look like this:

```
<input name="email" value="" data-np-autofill-type="username" data-np-uid="123"/>
```

I disabled the extension, and the error went away. This doesn't really resolve the problem though, as the form should work for people who use such extensions. I believe this is the same issue: https://github.com/remix-run/remix/issues/4822#issuecomment-1458566133

I guess the only way round this is to load the form up only once the page has fully loaded? Does anyone have any other ideas?

**Reply by sebpowell** [reply]
This also seems to be the same issue: https://stackoverflow.com/questions/75802996/how-to-handle-hydration-errors-caused-by-browser-extensions

Someone suggested some workarounds here, that seem to be what I mentioned above; the component should only be rendered client-side. 

https://javascript.plainenglish.io/how-to-solve-hydration-error-in-next-js-a50ec54bfc02

**Reply by icyJoseph** [reply]
If the extension is indeed modifying the DOM, then the current React hydration process flags the divergent HTML, but that in turns cascades into an error, and I think that's being handled in React@18.3.0 as the Remix issue suggests.

**Reply by sebpowell** [reply]
@icyJoseph – thanks! I'm going to try upgrading to 18.3.0 today and will report back. Hopefully that resolves it.

**Reply by icyJoseph** [reply]
Mmm not sure that's entirely a good idea... There's no such thing as 18.3.0, yet, but incremental releases (toward 18.3.0) they do, I'd say, since this is beta technology, you better own to it, and live with this issue until a proper release comes around.

**Reply by sebpowell** [reply]
@icyJoseph – ah sorry, I thought it had been released. In that case, I'll try what this guy suggested here: https://javascript.plainenglish.io/how-to-solve-hydration-error-in-next-js-a50ec54bfc02 – seems like it might be the best solution for now?
